File: src/main/java/carelog/carelog/user/app/UserService.java
package carelog.carelog.user.app;

import carelog.carelog.auth.app.CustomUserDetails;
import carelog.carelog.user.web.dto.CustomerCreateRequest;
import carelog.carelog.user.web.dto.ManagerCreateRequest;
import carelog.carelog.user.web.dto.UserResponse;
import carelog.carelog.user.web.dto.UserUpdateRequest;

import java.util.List;
import java.util.UUID;

public interface UserService {
    UserResponse createManager(ManagerCreateRequest request);
    UserResponse createCustomer(CustomerCreateRequest request, CustomUserDetails userDetails);
    UserResponse findUserByUserId(String userId);
    UserResponse findUserByEmail(String email);
    UserResponse updateUser(UUID publicId, UserUpdateRequest request);
    void deleteUser(UUID publicId);

    // ----- Customer 조회 -----
    List<UserResponse> findAllCustomers(String name);
}

File: src/main/kotlin/carelog/carelog/user/app/UserService.kt
package carelog.carelog.user.app

import carelog.carelog.auth.app.CustomUserDetails
import carelog.carelog.user.web.dto.CustomerCreateRequest
import carelog.carelog.user.web.dto.ManagerCreateRequest
import carelog.carelog.user.web.dto.UserResponse
import carelog.carelog.user.web.dto.UserUpdateRequest
import java.util.UUID

interface UserService {
    fun createManager(request: ManagerCreateRequest): UserResponse
    fun createCustomer(request: CustomerCreateRequest, userDetails: CustomUserDetails): UserResponse
    fun findUserByUserId(userId: String): UserResponse
    fun findUserByEmail(email: String): UserResponse
    fun updateUser(publicId: UUID, request: UserUpdateRequest): UserResponse
    fun deleteUser(publicId: UUID)
    fun findAllCustomers(name: String?): List<UserResponse>
}
